include/aws/http/private/h1_stream.h

```diff
@@ -55,6 +55,9 @@ struct aws_h1_stream {
     /* Buffer for incoming data that needs to stick around. */
     struct aws_byte_buf incoming_storage_buf;
 
+    /* Keep the underlying message alive */
+    struct aws_http_message *http_message;
```

Comment:
remove this. we didn't used to keep a pointer to the request from the stream, so we don't need to add one now

but we DO need to do reference counting on aws_h1_encoder_message->body